README.md

```diff
@@ -13,13 +18,171 @@ idseq_dag examples/host_filter_dag.json
 
 ```
 
+` idseq_dag --help ` for more options
+
 ## Test
 
 ```
 cd idseq-dag; python3 -m unittest
 
 ```
 
+or `python3 -m unittest tests/<module_file> ` for testing individual modules.
+
+## DAG Execution Details
+
+### Composing an example dag json file
+There are four basic elements of an IdSeq Dag
+
+ - **output_dir_s3**: the s3 directory where the output files will be copied to.
+ - **targets**: the outputs that are to be generated through dag execution. Each target consists of a list of files that will be copied to *output_dir_s3*
+ - **steps**: the steps that will be exeucuted in order to generate the targets. For each step, the following attributes can be specified:
+   * *in*: the input targets
+   * *out*: the output target
+   * *module*: name of python module
+   * *class*: name of python class that inherits [PipelineStep](idseq_dag/engine/pipeline_step.py)
+   * *additional_files*: additional S3 files required for dag execution, i.e. reference files.
```

Comment:
Just curious: why arent these input  targets themselves? Is it because they are shared?
The node could be more generic in that the a step could be loading these files. That node would be treated like any other step.

Reply:
because they are not generated from the pipeline and they don't trigger the start of the run function. But it's a design choice. we can put these in targets and call them given. I just feel separating the input/output and references is cleaner.